engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -328,6 +328,18 @@ namespace dmGameObject
         regist->m_Collections.SetSize(0);
     }
 
+    HCollection GetCollectionByHash(HRegister regist, dmhash_t socket_name)
+    {
+        uint32_t collection_count = regist->m_Collections.Size();
+        for (uint32_t i = 0; i < collection_count; ++i)
+        {
+            Collection* collection = regist->m_Collections[i];
+            if (collection->m_NameHash == socket_name)
+                return collection->m_HCollection;
+        }
+        return 0;
+    }
+
```

Comment:
I don't particularly _like_ this way, but since the solution has no context (all messages are sent to the generic `@physics:` socket, and flushed by all physics worlds), we need a way to get our context back (i.e. the calling collection).